homeassistant/components/ambient_network/config_flow.py

```diff
@@ -0,0 +1,154 @@
+"""Config flow for the Ambient Weather Network integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioambient import OpenAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LOCATION,
+    CONF_LONGITUDE,
+    CONF_MAC,
+    CONF_RADIUS,
+    UnitOfLength,
+)
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    LocationSelector,
+    LocationSelectorConfig,
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+from homeassistant.util.unit_conversion import DistanceConverter
+
+from .const import API_STATION_INDOOR, API_STATION_INFO, API_STATION_MAC_ADDRESS, DOMAIN
+from .helper import get_station_name
+
+CONF_USER = "user"
+CONF_STATION = "station"
+
+# One mile
+CONF_RADIUS_DEFAULT = 1609.34
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for the Ambient Weather Network integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Construct the config flow."""
+
+        self._longitude = 0.0
+        self._latitude = 0.0
+        self._radius = 0.0
+        self._stations: list[dict[str, Any]] = []
+
+    async def async_step_user(
+        self,
+        user_input: dict[str, Any] | None = None,
+        errors: dict[str, str] | None = None,
+    ) -> FlowResult:
+        """Handle the initial step to select the location."""
+
+        if user_input:
+            self._latitude = user_input[CONF_LOCATION][CONF_LATITUDE]
+            self._longitude = user_input[CONF_LOCATION][CONF_LONGITUDE]
+            self._radius = user_input[CONF_LOCATION][CONF_RADIUS]
+
+            client: OpenAPI = OpenAPI()
+            self._stations = await client.get_devices_by_location(
+                self._latitude,
+                self._longitude,
+                radius=DistanceConverter.convert(
+                    self._radius,
+                    UnitOfLength.METERS,
+                    UnitOfLength.MILES,
+                ),
```

Comment:
Interesting choice in `aioambient` to use miles

Reply:
It's not `aioambient`. It's the ambient API itself that requires miles.

Reply:
No matter what, I think it's an interesting choice